src/main/java/ArraySortedBag.java

```diff
@@ -216,7 +217,15 @@ public boolean equals(Object o) {
             return false;
         }
         ArraySortedBag<?> that = (ArraySortedBag<?>) o;
-        return Arrays.equals(this.bag, 0, this.rear, that.bag, 0, that.rear);
+        if (this.size() != that.size()) {
+            return false;
+        }
+        for (int i = 0; i < this.size(); i++) {
+            if (this.bag[i].compareTo((T) that.bag[i]) != 0) {
```

Comment:
Required a cast. 

The alternative was to cast the collection to be `ArraySortedBag<T>`. Perhaps one is more preferred over the other?